packages/backend/src/core/UptimeUpdater.ts

```diff
@@ -0,0 +1,95 @@
+import { Logger, TaskQueue } from '@l2beat/common'
+import { ProjectUptime } from '@l2beat/config'
+import { EthereumAddress } from '@l2beat/types'
+
+import { ProjectInfo } from '../model'
+import { ApiMonitor } from '../peripherals/uptime/ApiMonitor'
+import { RpcMonitor } from '../peripherals/uptime/RpcMonitor'
+import { Clock } from './Clock'
+
+export class UptimeUpdater {
+  private taskQueue = new TaskQueue<void>(() => this.update(), this.logger)
+
+  constructor(
+    private rpcMonitor: RpcMonitor,
+    private apiMonitor: ApiMonitor,
+    private clock: Clock,
+    private projects: ProjectInfo[],
+    private logger: Logger,
+  ) {
+    this.logger = this.logger.for(this)
+  }
+
+  start() {
+    this.logger.info('Started')
+    return this.clock.onNewMinute(() => {
```

Comment:
I think that it's better not to add new functionality to the Clock because the idea of a delay makes no sense for this real time scenario. I'd just use setInterval here directly.